src/XRP/shared/gateway-balances.ts

```diff
@@ -0,0 +1,76 @@
+import { XrpUtils } from 'xpring-common-js'
+import { XrpError, XrpErrorType } from '.'
+import { GatewayBalancesResponse } from './rippled-json-rpc-schema'
+
+/**
+ * Represents an amount of an Issued Currency.
+ */
+export interface IssuedCurrencyValue {
+  readonly currency: string
+  readonly value: string
+}
+
+/**
+ * Represents information about a gateway's issued currency balances on the XRPL, as of the most recent validated ledger version.
+ * @see https://xrpl.org/gateway_balances.html
+ */
+export default interface GatewayBalances {
+  /**
+   * The address of the account that issued the balances, encoded as an X-Address.
+   * @see https://xrpaddress.info/
+   */
+  readonly account: string
+
+  /**
+   * (Optional, omitted if empty) Total amounts held that are issued by others.
+   * In the recommended configuration, the issuing address should have none.
+   */
+  readonly assets?: { [account: string]: IssuedCurrencyValue[] }
+
+  /**
+   * (Optional, omitted if empty) Amounts issued to the excluded addresses from the request.
+   * The keys are addresses and the values are arrays of currency amounts they hold.
+   */
+  readonly balances?: { [account: string]: IssuedCurrencyValue[] }
+
+  /**
+   * (Optional, omitted if empty) Total amounts issued to addresses not excluded, as a map of currencies to the total value issued.
+   */
+  readonly obligations?: { [currencyCode: string]: string }
+
+  /** (May be omitted) The ledger index of the ledger version that was used to generate this response. */
+  readonly ledgerHash?: string | undefined
+}
+
+export function gatewayBalancesFromResponse(
+  gatewayBalancesResponse: GatewayBalancesResponse,
+): GatewayBalances {
+  if (gatewayBalancesResponse.result.validated === false) {
```

Comment:
it's pretty idiomatic to just do 
```
if (!gatewayBalancesResponse.result.validated)
```

for bools

Reply:
duh i'm such a newb

Reply:
although I guess my subconscious rationale wanted to make it more clear that this IS a bool since we're always checking all sorts of truthy and falsey things in JS, but I will definitely change this